### PR TITLE
ssh cli support for kubernetes

### PR DESCRIPTION
**Pull request by nsinkov**
## Changes proposed in this PR

use plugin to ssh to kubernetes

## Why are we making these changes?

we can't use ssh directly when a job is scheduled on kubernetes - we need to use kubectl

### CODE REVIEW

**Comment by dposada** [comment]
Will merge when green

File: cli/cook/subcommands/ssh.py [hunk 1]
```diff
@@ -1,19 +1,38 @@
...
-    os.execlp(command, 'ssh', '-t', hostname, f'cd "{sandbox_dir}" ; bash')
+    compute_cluster = instance["compute-cluster"]
+    compute_cluster_type = compute_cluster["type"]
+    compute_cluster_name = compute_cluster["name"]
+    if compute_cluster_type == "kubernetes":
+        kubectl_exec_to_instance_fn = plugins.get_fn('kubectl-exec-to-instance', kubectl_exec_to_instance)
+        cook_cluster_settings = http.get(cluster, 'settings', params={}).json()
+        compute_cluster_config = next(c for c in (s['config'] for s in cook_cluster_settings['compute-clusters']) if
+                                      c['compute-cluster-name'] == compute_cluster_name)
+        kubectl_exec_to_instance_fn(instance["task_id"], compute_cluster_config)
```
Comment (dposada) [inline cli/cook/subcommands/ssh.py]: Can we write an integration test for this?

File: cli/cook/subcommands/ssh.py [hunk 2]
```diff
@@ -1,19 +1,38 @@
...
 
 
-def ssh_to_instance(instance, sandbox_dir):
-    """Attempts to ssh (using os.execlp) to the Mesos agent corresponding to the given instance."""
+def kubectl_exec_to_instance(instance_uuid, _):
+    os.execlp('kubectl', 'kubectl',
+              'exec',
+              '-c', os.getenv('COOK_CONTAINER_NAME_FOR_JOB', 'required-cook-job-container'),
+              '-it', instance_uuid,
+              '--', '/bin/bash')
```
Comment (dposada) [inline cli/cook/subcommands/ssh.py]: Have we tested that this works?

File: cli/cook/subcommands/ssh.py (hunk 1 above)
Comment (nsinkov) [reply cli/cook/subcommands/ssh.py]: what about the cli tests?

Comment (dposada) [reply cli/cook/subcommands/ssh.py]: Ah, forgot the CLI tests don't currently support specifying a docker image to submit with. Until we solve that, not sure how much sense it makes to write a test for this.

**Comment by dposada** [comment]
Will merge when green

File: cli/cook/subcommands/ssh.py (hunk 2 above)
Comment (nsinkov) [reply cli/cook/subcommands/ssh.py]: yes

**Pull request merged**